src/main/java/org/craftercms/studio/impl/v2/utils/git/GitCli.java

```diff
@@ -175,10 +176,10 @@ private void handleErrorExitValue(String directory, GitCliOutputExceptionResolve
 			.orElse(new GitCliOutputException(exitValue, errorMessage));
 	}
 
-	private void handleProcessTimeout(Process p, String directory, InputStream processInputStream, InputStream processErrorStream) throws IOException {
+	private void handleProcessTimeout(Process p, String directory, File stdOutFile, File stdErrFile) throws IOException {
 		// Read available bytes, avoiding blocking
-		String stdOutput = new String(processInputStream.readNBytes(processInputStream.available()));
-		String errorOutput = new String(processErrorStream.readNBytes(processErrorStream.available()));
+		String stdOutput = Files.readString(stdOutFile.toPath());
```

Comment:
This can read until it exhausts all memory. Can we read the first x bytes for logging purposes and stop? Should we just log the path where the output file exists for the admins to read the actual temp file and see the logs? Any other path other than reading an unknown-size file into RAM?